### PR TITLE
feat: initail version of QB to expr

### LINKED ISSUE #3028: Research on QB languge to expr language for pipelines
Check if we can enable conversion of QB language to expr language

### PR DESCRIPTION
**Pull request by nityanandagohain**
Fixes #3028

### CODE REVIEW

File: pkg/query-service/queryBuilderToExpr/queryBuilderToExpr.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	"go.uber.org/zap"
+)
+
+var operatorsSupported = map[string]struct{}{
+	"in":         {},
+	"contains":   {},
+	"startsWith": {},
+	"endsWith":   {},
+	"matches":    {},
+}
```
Comment (srikanthccv) [inline pkg/query-service/queryBuilderToExpr/queryBuilderToExpr.go:19]: This is not used anywhere.

File: pkg/query-service/queryBuilderToExpr/queryBuilderToExpr.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		return ""
+	}
+}
+
+func quoteEscapedString(str string) string {
+	// https://clickhouse.com/docs/en/sql-reference/syntax#string
+	str = strings.ReplaceAll(str, `\`, `\\`)
+	str = strings.ReplaceAll(str, `'`, `\'`)
+	return str
+}
```
Comment (srikanthccv) [inline pkg/query-service/queryBuilderToExpr/queryBuilderToExpr.go:137]: Do the same ClickHouse escaping rules apply here?

File: pkg/query-service/queryBuilderToExpr/queryBuilderToExpr_test.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		Expr: "'key' not in attributes",
+	},
+	{
+		Name: "Multi filter",
+		Query: &v3.FilterSet{Operator: "AND", Items: []v3.FilterItem{
+			{Key: v3.AttributeKey{Key: "key", DataType: v3.AttributeKeyDataTypeString, Type: v3.AttributeKeyTypeTag}, Value: 10, Operator: "<="},
+			{Key: v3.AttributeKey{Key: "body", DataType: v3.AttributeKeyDataTypeString, Type: v3.AttributeKeyTypeUnspecified, IsColumn: true}, Value: "[0-1]+regex$", Operator: "nregex"},
+			{Key: v3.AttributeKey{Key: "key", DataType: v3.AttributeKeyDataTypeString, Type: v3.AttributeKeyTypeTag}, Operator: "nexists"},
+		}},
+		Expr: "attributes.key <= 10 and body not matches '[0-1]+regex$' and 'key' not in attributes",
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply pkg/query-service/queryBuilderToExpr/queryBuilderToExpr_test.go:130]: We only support either `AND` or `OR` b/w filter items. Such composite expression with a mix of `AND` and `OR` is not supported for now.

**Review by srikanthccv — COMMENTED** [review]
Overall LGTM, some minor comments.

File: pkg/query-service/queryBuilderToExpr/queryBuilderToExpr.go (hunk 2 above)
Comment (nityanandagohain) [reply pkg/query-service/queryBuilderToExpr/queryBuilderToExpr.go]: have added changes with test cases, it needed some changes

File: pkg/query-service/queryBuilderToExpr/queryBuilderToExpr_test.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+func TestParse(t *testing.T) {
+	for _, tt := range testCases {
+		Convey(tt.Name, t, func() {
+			x, err := Parse(tt.Query)
+			if tt.ExpectError {
+				So(err, ShouldNotBeNil)
+			} else {
+				So(err, ShouldBeNil)
+				fmt.Println(x)
```
Comment (srikanthccv) [inline pkg/query-service/queryBuilderToExpr/queryBuilderToExpr_test.go:160]:
```suggestion
```

**Comment by srikanthccv** [comment]
Resolve the conflicts

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**